python/tests/test_ownership_dataset.py

```diff
@@ -0,0 +1,275 @@
+    assert sc.identical(a, sc.array(dims=['x'], values=[-100, -200]))
+
+    # TODO intentional?
+    # Assignments overwrite data but not coords.
+    da.data = sc.array(dims=['x'], values=[11, 22], unit='m')
+    da.coords['x'] = sc.array(dims=['x'], values=[3, 4], unit='s')
+    da.attrs['a'] = sc.array(dims=['x'], values=[300, 400])
+    assert sc.identical(
+        da,
+        sc.DataArray(
+            sc.array(dims=['x'], values=[11, 22], unit='m'),
+            coords={'x': sc.array(dims=['x'], values=[3, 4], unit='s')},
+            attrs={'a': sc.array(dims=['x'], values=[300, 400])}))
+    assert sc.identical(v, sc.array(dims=['x'], values=[11, 22], unit='m'))
+    assert sc.identical(c, sc.array(dims=['x'], values=[-1, -2], unit='J'))
+    assert sc.identical(a, sc.array(dims=['x'], values=[-100, -200]))
+
+
+def test_own_darr_copy():
+    # Depth of copy can be controlled.
+    da, _, c, a, m = make_data_array()
+    da_copy = copy(da)
+    da_deepcopy = deepcopy(da)
+    da_methcopy = da.copy(deep=False)
+    da_methdeepcopy = da.copy(deep=True)
+    da['x', 0] = -10
+    da.data['x', 1] = -20
+    da.coords['x']['x', 0] = -1
+    da.attrs['a']['x', 0] = -100
+    da.masks['m']['x', 0] = False
+    c['x', 1] = -2
+    a['x', 1] = -200
+    m['x', 1] = True
+    da.unit = 'kg'
+    da.coords['x'].unit = 'J'
+
+    modified = sc.DataArray(
+        sc.array(dims=['x'], values=[-10, -20], unit='kg'),
+        coords={'x': sc.array(dims=['x'], values=[-1, -2], unit='J')},
+        attrs={'a': sc.array(dims=['x'], values=[-100, -200])},
+        masks={'m': sc.array(dims=['x'], values=[False, True])})
+    assert sc.identical(da, modified)
+    assert sc.identical(da_copy, modified)
+    assert sc.identical(da_deepcopy, make_data_array()[0])
+    assert sc.identical(da_methcopy, modified)
+    assert sc.identical(da_methdeepcopy, make_data_array()[0])
+
+
+def test_own_dset_set_access_through_dataarray():
+    # The DataArray is shared.
+    dset = sc.Dataset()
+    da, *_ = make_data_array()
+    dset['da1'] = da
+
+    dset['da1']['x', 0] = -10
+    dset['da1'].attrs['a']['x', 0] = -100
+    dset['da1'].masks['m']['x', 0] = False
+    da['x', 1] = -20
+    da.coords['x']['x', 1] = -2
+    da.attrs['a']['x', 1] = -200
+    da.masks['m']['x', 1] = True
+    dset['da1'].unit = 'kg'
+
+    expected = sc.DataArray(
+        sc.array(dims=['x'], values=[-10, -20], unit='kg'),
+        coords={'x': sc.array(dims=['x'], values=[1, -2], unit='s')},
+        attrs={'a': sc.array(dims=['x'], values=[-100, -200])},
+        masks={'m': sc.array(dims=['x'], values=[False, True])})
+    assert sc.identical(dset, sc.Dataset({'da1': expected}))
+    assert sc.identical(da, expected)
+
+
+def test_own_dset_set_access_through_scalar_slice():
+    # The DataArray is shared.
+    dset = sc.Dataset()
+    da, *_ = make_data_array()
+    dset['da1'] = da
+
+    dset['x', 0]['da1'].value = -10
+    # TODO allows modification of coord through attrs
```

Comment:
What is this referring to?

Reply:
Oh, you mean it is missing a readonly flag, since the dataset item access implies it is shared?